src/send/Send.tsx

```diff
@@ -135,10 +137,36 @@ function Send({ route }: Props) {
       )
 
       setSelectedRecipient(recipient)
+
+      if (defaultTokenOverride) {
+        navigate(Screens.SendAmount, {
+          defaultTokenOverride,
+          forceTokenAddress,
+          recipient,
+          isOutgoingPaymentRequest,
+          origin: isOutgoingPaymentRequest ? SendOrigin.AppRequestFlow : SendOrigin.AppSendFlow,
+        })
+      } else {
+        setShowCurrencyPicker(true)
+      }
     },
     [isOutgoingPaymentRequest, searchQuery]
   )
 
+  const onTokenSelected = (token: string) => {
+    setShowCurrencyPicker(false)
+    if (!recipient) {
```

Comment:
What case is this for? Shouldn't this only be callable after a recipient has been selected?

Reply:
This is 1: to be defensive, since _technically_ there aren't any compile-time guarantees that this will always be defined, and 2. to appease the Typescript lords, since we can't pass `string | undefined` as a value for `recipient` to the `SendAmount` screen.